src/core/p5.Graphics.js

```diff
@@ -60,6 +60,13 @@ p5.Graphics = class extends p5.Element {
 
     if (r === constants.WEBGL) {
       this._renderer = new p5.RendererGL(this.canvas, this, false);
+      const gl = this._renderer.GL;
+      const maxTextureSize = gl.getParameter(gl.MAX_TEXTURE_SIZE);
+      const maxAllowedPixelDimensions = Math.floor(
+        maxTextureSize / this._pixelDensity
+      );
+      w = Math.min(w, maxAllowedPixelDimensions);
+      h = Math.min(h, maxAllowedPixelDimensions);
```

Comment:
We might want to log a warning here (and in p5.Framebuffer.js) if the width/height we end up picking is different than what the user requested.